Wikipedia/Code/NSUserDefaults+WMFExtensions.swift

```diff
@@ -79,5 +79,35 @@ extension NSUserDefaults {
             return false
         }
     }
+    
+    public func wmf_setAppInstallDateIfNil(date: NSDate) {
+        let previous = self.wmf_appInstallDate()
+        
+        if previous == nil {
+            self.setObject(date, forKey: "AppInstallDate")
+            self.synchronize()
+        }
+    }
+    
+    public func wmf_appInstallDate() -> NSDate? {
+        if let date = self.objectForKey("AppInstallDate") as? NSDate {
```

Comment:
@bgerstle I did that at first and it complained - it wanted me to wrap everything in a force unwrap